testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeWriteDatabricksCompatibility.java

```diff
@@ -286,6 +305,37 @@ public void testInsertingIntoDatabricksTableWithAddedNotNullConstraint()
         }
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_OSS, PROFILE_SPECIFIC_TESTS})
+    public void testTrinoVacuumRemoveChangeDataFeedFiles()
+    {
+        String tableName = "test_vacuum_ignore_cdf_" + randomTableSuffix();
+        String directoryName = "databricks-compatibility-test-" + tableName;
+        String changeDataPrefix = directoryName + "/_change_data";
+
+        onDelta().executeQuery("CREATE TABLE default." + tableName + " (a INT) " +
+                "USING DELTA " +
+                "LOCATION '" + ("s3://" + bucketName + "/" + directoryName) + "'" +
+                "TBLPROPERTIES (delta.enableChangeDataFeed = true)");
+
+        try {
+            // Executing INSERT and DELETE statements creates _change_data directory
+            onDelta().executeQuery("INSERT INTO " + tableName + " VALUES (1)");
+            onDelta().executeQuery("UPDATE " + tableName + " SET a = 2");
+
+            Assertions.assertThat(s3.listObjectsV2(bucketName, changeDataPrefix).getObjectSummaries()).hasSize(1);
+
+            // Vacuum procedure should remove files in _change_data directory
+            // https://docs.delta.io/2.1.0/delta-change-data-feed.html#change-data-storage
+            onTrino().executeQuery("SET SESSION delta.vacuum_min_retention = '0s'");
+            onTrino().executeQuery("CALL delta.system.vacuum('default', '" + tableName + "', '0s')");
+
+            Assertions.assertThat(s3.listObjectsV2(bucketName, changeDataPrefix).getObjectSummaries()).hasSize(0);
```

Comment:
Should or shouldn't remove?

Reply:
My gut feeling is that CDF files which are referenced from transaction log versions which are retained should also be retained. 

So in this case, we vacuum everything except for the most recent version of the table, there should still be one CDF file left after the vacuum.

Does that seem like the right behavior?

Reply:
Maybe we could see what databricks is doing and do the same thing? Or have you checked it already?

Reply:
> My gut feeling is that CDF files which are referenced from transaction log versions which are retained should also be retained.

I just tried this out on Databricks 10.4 and this is not what they do. If you have the retention time set to zero they delete all of the CDF files. 

I guess this is fine then

Reply:
> Or have you checked it already?

Yes, I confirmed Databricks behavior before sending this PR. Added another test which runs on Databricks just in case.

Reply:
> I just tried this out on Databricks 10.4 and this is not what they do. If you have the retention time set to zero they delete all of the CDF files.

😮 
so they treat CDF files just like any other untracked files?
 @vkorukanti can you please confirm?